### PR TITLE
Add workflows and update heroku config

### PR DESCRIPTION
**Pull request by smart-alek**
Updating configuration to:

 - use workflows
- use updated, simpler Heroku deploy

**Note:** `HEROKU_APP_NAME` will have to be set as an environment variable in the project's settings.

### CODE REVIEW

**Comment by smart-alek** [comment]
Follow-up PR for [this documentation change](https://github.com/circleci/circleci-docs/pull/2315)

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 version: 2
 jobs:
-  build:
+  build-job:
```
Comment (ndintenfass) [inline .circleci/config.yml]: Curious: why `-job`?

Comment (smart-alek) [reply .circleci/config.yml]: Was just maintaining Phil and Jose's formatting decision from their example. I have no strong feelings about this, actually — perhaps it's needlessly explicit?

Comment (technomancy) [reply .circleci/config.yml]: I don't remember intentionally adding `-job` to the example Jose and I made; must have pulled it in by accident. I agree it's not needed; just a bit redundant at worst, not a big deal.

**Pull request merged**